habitat/tasks/nav/nav_task.py

```diff
@@ -363,6 +444,147 @@ def update_metric(self, episode, action):
+        self._cell_scale = (
+            self._coordinate_max - self._coordinate_min
+        ) / self._map_resolution[0]
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "top_down_map"
+
+    def _check_valid_nav_point(self, point: List[float]):
+        self._sim.is_navigable(point)
+
+    def get_original_map(self, episode):
+        top_down_map = maps.get_topdown_map(
+            self._sim,
+            self._map_resolution,
+            self._num_samples,
+            self._draw_border,
+        )
+
+        range_x = np.where(np.any(top_down_map, axis=1))[0]
+        range_y = np.where(np.any(top_down_map, axis=0))[0]
+
+        self._ind_x_min = range_x[0]
+        self._ind_x_max = range_x[-1]
+        self._ind_y_min = range_y[0]
+        self._ind_y_max = range_y[-1]
+
+        if self._draw_source_and_target:
+            # mark source point
+            s_x, s_y = maps.to_grid(
+                episode.start_position[0],
+                episode.start_position[2],
+                self._coordinate_min,
+                self._coordinate_max,
+                self._map_resolution,
+            )
+            point_padding = 2 * int(
+                np.ceil(self._map_resolution[0] / MAP_THICKNESS_SCALAR)
+            )
+            top_down_map[
+                s_x - point_padding : s_x + point_padding + 1,
+                s_y - point_padding : s_y + point_padding + 1,
+            ] = 4
+
+            # mark target point
+            t_x, t_y = maps.to_grid(
+                episode.goals[0].position[0],
+                episode.goals[0].position[2],
+                self._coordinate_min,
+                self._coordinate_max,
+                self._map_resolution,
+            )
+            top_down_map[
+                t_x - point_padding : t_x + point_padding + 1,
+                t_y - point_padding : t_y + point_padding + 1,
+            ] = 6
+
+        return top_down_map
+
+    def reset_metric(self, episode):
+        self._step_count = 0
+        self._metric = None
+        self._top_down_map = self.get_original_map(episode)
+        agent_position = self._sim.get_agent_state().position
+        a_x, a_y = maps.to_grid(
+            agent_position[0],
+            agent_position[2],
+            self._coordinate_min,
+            self._coordinate_max,
+            self._map_resolution,
+        )
+        self._previous_xy_location = (a_y, a_x)
+
+    def update_metric(self, episode, action):
+        self._step_count += 1
+        house_map, map_agent_x, map_agent_y = self.update_map(
+            self._sim.get_agent_state().position
+        )
+
+        house_map = house_map[
+            self._ind_x_min
+            - self._grid_delta : self._ind_x_max
+            + self._grid_delta,
+            self._ind_y_min
+            - self._grid_delta : self._ind_y_max
+            + self._grid_delta,
+        ]
+
+        self._metric = house_map
```

Comment:
In `habitat.Measure` metric is a number. Maybe that's good to switch it to something more general.

Reply:
I get what you mean, but in the interface it doesn't seem to specify the type. What if we made this a Sensor or an Info?

Reply:
> I get what you mean, but in the interface it doesn't seem to specify the type. What if we made the top down map a sensor or an Info?
My bad, ignore this comment please.